go/tendermint/apps/staking/staking.go

```diff
@@ -158,40 +158,42 @@ func (app *stakingApplication) InitChain(ctx *abci.Context, request types.Reques
 		return errors.Wrap(err, "staking/tendermint: failed to add common pool")
 	}
 
-	var ups ledgerUpdates
+	// Ledger.
+	var ups []ledgerUpdate
 	for k, v := range st.Ledger {
 		var id signature.PublicKey
-		_ = id.UnmarshalBinary(k[:])
+		id.FromMapKey(k)
 
-		if !v.GeneralBalance.IsValid() {
+		if !v.General.Balance.IsValid() {
 			app.logger.Error("InitChain: invalid genesis general balance",
 				"id", id,
-				"general_balance", v.GeneralBalance,
+				"general_balance", v.General.Balance,
 			)
 			return errors.New("staking/tendermint: invalid genesis general balance")
 		}
-		if !v.EscrowBalance.IsValid() {
+		if !v.Escrow.Balance.IsValid() {
 			app.logger.Error("InitChain: invalid genesis escrow balance",
 				"id", id,
-				"escrow_balance", v.EscrowBalance,
+				"escrow_balance", v.Escrow.Balance,
 			)
 			return errors.New("staking/tendermint: invalid genesis escrow balance")
 		}
-
-		account := &ledgerEntry{
-			GeneralBalance:  v.GeneralBalance,
-			EscrowBalance:   v.EscrowBalance,
-			DebondStartTime: v.DebondStartTime,
-			Nonce:           v.Nonce,
+		if !v.Escrow.DebondingShares.IsZero() {
```

Comment:
Whowa, is this an ok assumption to make?  I don't think it is...

Reply:
Fixed so dump/restore now also dumps and restores unbonding delegations.